### ISSUE TITLE
autoFixOnSave Throws Cursor to End of File

### ISSUE THREAD

**Issue by bartlewis** [Microsoft/vscode-eslint#341]
After updating to VSCode 1.18.0 this morning, each time that ESLint runs, the cursor gets thrown to the bottom of the file. This seems to be related to the prettier plugin/rule.

**Comment by dbaeumer** [comment]
This doesn't reproduce for me with 1.18 and eslint 1.4.3. I guess this is caused by a plugin different than eslint.

Can you please disable all extension except eslint and try to reproduce. If you can pleas provide a GitHub repository I can clone that demos the problem with steps to reproduce.

**Comment by bartlewis** [comment]
Thanks @dbaeumer. Working on putting something together, but have not yet been able to isolate the issue.

**Comment by joelmukuthu** [comment]
I'm also experiencing this issue and it seems to happen when eslint is integrated with prettier. I disabled all other plugins besides eslint and trimmed down my vscode settings to only the ones below.

The cursor jumps to the end of the file on saving (with Ctrl+S) if there are **formatting** (i.e. prettier) errors to be fixed. If there are no lint errors, or there are other errors that are not triggered by prettier, the cursor stays at the same position.

Minimal setup:
```bash
npm install eslint prettier eslint-plugin-pretter eslint-config-prettier
```
.eslintrc.json:
```json
{
  "plugins": [
    "prettier"
  ],
  "extends": [
    "prettier"
  ],
  "parserOptions": {
    "ecmaVersion": 2017
  },
  "rules": {
    "prettier/prettier": "error"
  }
}
```
sample file:
```js
// foo.js
const foo = {
  foo                  : 'foo' // this will trigger a formatting error 
};
```
vscode settings:
```json
{
    "files.autoSave": "off",
    "eslint.autoFixOnSave": true
}
```
vscode-eslint version: 1.4.7
vscode version: 1.20.1

**Comment by dbaeumer** [comment]
I can't reproduce this given the provided steps:

![cast](https://user-images.githubusercontent.com/1931590/36968123-e6f0c4e8-2061-11e8-9e73-20391e656b2a.gif)

**Comment by joelmukuthu** [comment]
Hmm.. I'm on my work computer and now I can't either. I'll try again when I get to my home computer. One thing though, if you have the cursor in between `'foo'`:
```js
// foo.js
const foo = {
  foo                  : 'foo' // put the cursor on column 28 here
};
```
then save the file, the cursor jumps to the end of the line, perhaps because the `'foo'` gets changed to `"foo"`. is that expected or is it also the bug manifesting in a different way?

**Comment by dbaeumer** [comment]
This is somehow expected since the ESLint formatter replaces a wide range of text making it hard to keep the cursor stable. However if you see something really wired feel free to open a seperate issue.

**Comment by joelmukuthu** [comment]
Okay. I'm still experiencing this issue (jumping to end of file) every now and then but I haven't managed to create a minimal test case. I'll post back here if I manage

**Comment by neo21670** [comment]
+1 also experiencing this, even for minor modifications/fixes.

**Comment by mellson** [comment]
I had the same problem with the [sort-imports](https://github.com/amatiasq/vsc-sort-imports) extension installed

**Comment by dbaeumer** [comment]
Can someone experiencing this provide me with a sample file or a GitHub repository I can clone.

**Comment by davidtn** [comment]
No, this is a real problem. I'm editing Go code and on save that causes a reformat then the cursor gets thrown to the end of file.

Interestingly this only started to happen when I updated to the latest version of Go (which causes the formatting tools to be rebuilt I think)

**Comment by dbaeumer** [comment]
@davidtn can you please report this then to the Go extension. This is about the ESLint extension.
